spec/at-rules/for.md

```diff
@@ -22,38 +22,38 @@
 
 To execute a `@for` rule `rule`:
 
-* Let `from` be the result of evaluating the expression in `FromDeclaration`
-
-* Assert that `from` is a number
+* Let `from` be the result of evaluating the expression in `FromDeclaration`.
 
 * If `rule` has a `ToDeclaration`:
 
-  * Let `to` be the result of evaluating the expression in `ToDeclaration`
+  * Let `to` be the result of evaluating the expression in `ToDeclaration`.
 
-  * Let `exclusive` be `true`
+  * Let `exclusive` be `true`.
   
 * Otherwise:
 
-  * Let `to` be the result of evaluating the expression in `ThroughDeclaration`
+  * Let `to` be the result of evaluating the expression in `ThroughDeclaration`.
 
-  * Let `exclusive` be `false`
+  * Let `exclusive` be `false`.
   
-* Assert that `to` is a number
-
-* Assert that `from` and `to` are integers
-
-* If `from` is greater than `to`, set `direction` to `-1`. Otherwise, set `direction` to `1`
+* If `from` and `to` aren't integers, throw an error.
 
-* If `exclusive` is `false`, set `to` as the result of adding `direction` to `to`
+* If `from` is greater than `to`, set `direction` to `-1`. Otherwise, set
+  `direction` to `1`.
 
-* If `from` and `to` are equal, stop here
+* If `exclusive` is `false`, set `to` to `to + direction`.
 
-* Set `i` to `from`
+* Let `i` be `from`.
 
 * While `i` is not equal to `to`:
 
-  * Declare a variable with `rule`'s `VariableName` as name and value `i` in the `rule`'s `ForBlock`'s scope
+  * Let `scope` be a new [scope].
```

Comment:
do we actually create a new scope at each iteration ?

Reply:
@nex3 was there anything wrong with my previous proposal ? According to the spec for variables, the curly braces of the `ForBlock` are already introducing a new scope.

Reply:
I don't think it's actually detectable either way right now, but if we do ever add full closure semantics to Sass we'll want a closure to capture the *current* value of the iteration variable rather than the *final* value (that is, we'll want it to behave like `for`/`let` in JS and not like `for`/`var`). This text makes that explicit.